libs/platform/src/lib/approval-flow/helpers/data-source.ts

```diff
@@ -0,0 +1,110 @@
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import { DataProvider } from '@fundamental-ngx/platform/shared';
+import { delay, distinctUntilChanged, take } from 'rxjs/operators';
+import { ApprovalDataSource } from '../interfaces/approval-data-source';
+import { ApprovalProcess } from './../interfaces/approval-process';
+import { ApprovalUser } from './../interfaces/approval-user';
+import { ApprovalTeam } from './../interfaces/approval-team';
+import { ApprovalNode } from './../interfaces/approval-node';
+
+export class DefaultApprovalDataSource implements ApprovalDataSource {
```

Comment:
This introduce allot of complexity. 

AF Component will have one way of listening for data coming in from provider. This is why we have open() method in the DataSource Interface. 

I know that this AFDS requires two data sources, but we should follow the same semantics like other implementations.